### PR TITLE
M6 IMPL - PR #2 dest-hub spine end-to-end

### PR DESCRIPTION
**Pull request by Sidarthapati**
## Summary
Implements the M7 destination-hub path end-to-end and makes it as dynamic as the origin side: a landed parcel is sorted through a `hex → DA territory → delivery route` ladder into a **route bag** (a van loads the whole bag) or a **DA-territory bag** fallback (no van), and the per-parcel **`PARCEL_SORTED_FOR_DELIVERY`** feed becomes the real backing for M6's binder — replacing its buffer stub with no change to M6.

---

## What Changed
- **Destination sort ladder (`M7-D-012`):** new `SortService.resolveInbound` walks `dest_hex → DA territory (M3) → delivery route/loop (M6)`; opens a `ROUTE` bag when a van runs, falls back to a `DA_TERRITORY` bag otherwise — dynamically allocating a free stand from the shared pool, never per-hex.
- **New consolidation unit:** `delivery_bag` + `DeliveryBagService` (the inbound mirror of `flight_bag`/`BagService`); `bag_manifest` generalized to serve both directions (`direction` + `manifest_kind`, FK relaxed).
- **The M6 seam goes live:** `common.ParcelSortedForDeliveryEvent` carries M6's exact 6-field binder shape **plus additive** fields (`daTerritoryId`, `routePlanId`/`loopId`, `deliveryBagId`, `standNo`) that M6's tolerant reader ignores.
- **Real M3/M6 wiring:** `TerritoryPort` → `GridTerritoryAdapter` (over `GridService`), `DeliveryRoutePort` → `RoutingDeliveryRouteAdapter` (over `RoutePlanLifecycleService`); `grid`+`routing` added as hub deps (interface-only imports).
- **Same-city shortcut (§12)** and **hub-collect shelf (§8.2):** receiving derives arrival mode and routes VAN/SELF_DROP (same-city skips the flight) / AIRPORT / HUB_COLLECT accordingly.
- **Migrations** `V7_10`–`V7_12`; `findFreeStands` now excludes open bags of *both* kinds from the one shared stand pool.
- **M4 read seam extended:** `ShipmentInfo` surfaces `destTileId` (the destination hex).

---

## Why This Change?
M7's origin path (PR #1) was already dynamic, but the destination side was unbuilt and M6 was reading delivery parcels from a provisional buffer stub. This PR delivers the real dynamic destination hub and closes that seam: M6 now binds against M7's actual sort output, the van loads a whole route bag in one move, and a no-van city still functions via DA hub-collect — satisfying the "dynamic assignment everywhere" requirement with no static directory.

---

## Testing
- [x] Tested locally
- [x] Edge cases considered
- [x] No breaking changes introduced

### Test Evidence
`mvn clean install -pl common,orders,grid,routing,hub -am` — all

Tests run: 9  — com.oneday.hub.service.impl.BagServiceImplTest
Tests run: 5  — com.oneday.hub.service.impl.HubReceivingServiceImplTest
Tests run: 3  — com.oneday.hub.service.impl.SortServiceImplTest
Tests run: 4  — com.oneday.hub.service.impl.DeliveryBagServiceImplTest
Tests run: 5  — com.oneday.hub.events.HubEventPayloadJacksonTest
Tests run: 26, Failures: 0, Errors: 0, Skipped: 0   (hub module)
[INFO] BUILD SUCCESS

Key contract test (`m6TolerantReader_recoversCoreShape_ignoresAddtedForDeliveryEvent` JSON deserializes into M6's provisional`routing.events.payload.ParcelSortedForDeliveryEvent` with all 6 core fields recovered and additive fields silently ignored — proving the stub→real swap is invisible to M6's binder.

Migrations `V7_1…V7_12` applied in a throwaway schema (rolled bac
 stands=50
 manifest_cols=id,bag_id,flight_no,parcel_count,weight_grams,parct,direction,manifest_kind
 delbag_indexes=6
 ROLLBACK

---

### Areas Affected
- [x] Backend
- [ ] Frontend
- [x] Routing
- [ ] Infra
- [x] Database
- [ ] NA

### Modules Affected
- [ ] M1 — Auth
- [ ] M2 — Pricing
- [x] M3 — Grid
- [x] M4 — Orders
- [ ] M5 — Dispatch
- [x] M6 — Routing
- [x] M7 — Hub
- [ ] M8 — Barcode
- [ ] M9 — Airline
- [ ] M10 — SLA
- [ ] M11 — Exceptions
- [ ] NA

---

## Risks / Concerns
- **New compile-time deps `hub → grid, routing`.** Acyclic (grid←common; routing←grid,common; neither depends on hub) and interface-only imports per the
cross-module rule. Pulls routing's heavier deps onto hub's classp in hub-isolated tests.
- **Ports wire real (not stub-only).** M3/M6 are built, so this is the integration PR; if their query contracts shift, the adapters need updating. Unit tests
mock the ports so the suite is insulated.
- **`V7_12` alters PR #1's `bag_manifest`** (drops the flight-only FK, makes `flight_no` nullable). Forward-only and validated against Postgres; a local DB that
already applied PR #1 needs a clean/reseed.
- **ZONE-bag grouping (Q13) deferred:** enum + column exist, but pool exhaustion currently escalates via `NoFreeStandException` rather than grouping adjacent
territories — PR #3 owns back-pressure.

---

## Checklist
- [x] Code follows project conventions (package layout, no cross-module internal imports)
- [x] Self-review completed
- [x] Append-only audit trail preserved (no mutations to scan/manifest/role-change records)
- [x] No sensitive data or credentials exposed
- [x] Documentation updated if behaviour changed
- [x] Ready for review

### CODE REVIEW

File: common/src/main/java/com/oneday/common/kafka/events/hub/ParcelSortedForDeliveryEvent.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        UUID parcelId,
+        UUID cityId,
+        UUID destinationHexId,
+        LocalDate validDate,
+        Instant sortedAt,
+        Instant slaDeadline,
+        // ── additive (M6-tolerant) ──
+        UUID daTerritoryId,
+        UUID routePlanId,
+        UUID loopId,
```
Comment (Sidarthapati) [inline common/src/main/java/com/oneday/common/kafka/events/hub/ParcelSortedForDeliveryEvent.java]: What is loopId, have not seen it till now, neither is it a part of the db?

File: common/src/main/java/com/oneday/common/kafka/events/hub/ParcelSortedForDeliveryEvent.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        UUID deliveryBagId,
```
Comment (Sidarthapati) [inline common/src/main/java/com/oneday/common/kafka/events/hub/ParcelSortedForDeliveryEvent.java:32]: Same what is deliverybagid, same as bagid or different ? We already have parcelId

File: common/src/main/java/com/oneday/common/kafka/events/hub/DestSortCompleteEvent.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * Per parcel: destination sortation is complete (the parcel is staged on its delivery stand, §8.2).
+ * Consumed by M10 for the dest-hub leg's SLA accounting. Keyed by the parcel so per-parcel ordering
+ * holds; {@code wave} carries the operating day for roll-up.
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public record DestSortCompleteEvent(
+        UUID parcelId,
+        UUID cityId,
+        UUID hubId,
+        LocalDate wave,
```
Comment (Sidarthapati) [inline common/src/main/java/com/oneday/common/kafka/events/hub/DestSortCompleteEvent.java]: What is wave here, clarify please, is that loop index?

File: hub/src/main/java/com/oneday/hub/domain/DeliveryBag.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    @Column(name = "bag_date", nullable = false, updatable = false)
+    private LocalDate bagDate;
+
+    // Exactly one of loop_id / da_territory_id / zone_id is set, per bag_kind (the lazy-create key).
+    @Column(name = "route_plan_id", updatable = false)
+    private UUID routePlanId;
+
+    @Column(name = "loop_id", updatable = false)
+    private UUID loopId;
```
Comment (Sidarthapati) [inline hub/src/main/java/com/oneday/hub/domain/DeliveryBag.java]: Same question

File: hub/src/main/java/com/oneday/hub/dto/ReceiveResponse.java [hunk 5]
```diff
@@ -7,27 +7,56 @@
...
+                    s.bagKind() != null ? s.bagKind().name() : null,
+                    s.destHexId(), s.daTerritoryId(), s.routePlanId(), s.loopId());
+        }
+        // HUB_COLLECT shelf placement — no bag, no stand.
         return new ReceiveResponse(r.receiptId(), r.shipmentId(), r.shipmentRef(),
-                r.reconciled(), r.discrepancyType(),
-                s.sortKey(), s.bagId(), s.standId(), s.standNo(),
-                s.flightNo(), s.flightDate(), s.destHub(), s.bagCutoff());
+                r.reconciled(), r.discrepancyType(), "HUB_COLLECT",
+                null, null, null, null, null, null, null, null, null, null, null, null, null);
```
Comment (Sidarthapati) [inline hub/src/main/java/com/oneday/hub/dto/ReceiveResponse.java:60]: WHat is this Hub Collect shelf placement and when are we using this?

File: hub/src/main/java/com/oneday/hub/service/impl/DeliveryBagServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+        if (bag.getStatus() != DeliveryBagStatus.OPEN) {
+            throw new IllegalBagStateException("Delivery bag " + deliveryBagId + " is already " + bag.getStatus());
+        }
+        List<DeliveryBagItem> items = deliveryBagItemRepository
+                .findByDeliveryBagIdAndStatus(deliveryBagId, DeliveryBagItemStatus.STAGED);
+
+        BagManifest manifest = bagManifestRepository.save(BagManifest.builder()
+                .bagId(deliveryBagId)
+                .direction(SortDirection.INBOUND)
+                .manifestKind(ManifestKind.LOAD_LIST)
```
Comment (Sidarthapati) [inline hub/src/main/java/com/oneday/hub/service/impl/DeliveryBagServiceImpl.java:145]: What is load list?

File: hub/src/main/java/com/oneday/hub/service/impl/HubReceivingServiceImpl.java [hunk 7]
```diff
@@ -17,26 +21,36 @@
...
 
     private final ShipmentInfoPort shipmentInfoPort;
     private final InboundReceiptRepository inboundReceiptRepository;
+    private final DeliveryBagItemRepository deliveryBagItemRepository;
     private final SortService sortService;
+    private final HubEventProducer eventProducer;
     private final Clock clock;
 
     HubReceivingServiceImpl(ShipmentInfoPort shipmentInfoPort,
                             InboundReceiptRepository inboundReceiptRepository,
```
Comment (Sidarthapati) [inline hub/src/main/java/com/oneday/hub/service/impl/HubReceivingServiceImpl.java:44]: What is inboundrecietrepository?

File: hub/src/main/java/com/oneday/hub/service/impl/FlightBagServiceImpl.java [hunk 8]
```diff
@@ -68,7 +68,7 @@ private FlightBag createBag(OpenBagCommand cmd) {
         // Dynamic stand assignment: grab the next free shelf from the hub's shared stand pool. The
         // stand becomes "this flight's bag" only now — nothing is pre-mapped. Freed again on dispatch.
         Stand stand = standRepository
-                .findFreeStands(cmd.hubId(), StandStatus.OPEN, BagStatus.OPEN)
+                .findFreeStands(cmd.hubId(), StandStatus.OPEN, "AIRPORT_DOCK")
```
Comment (Sidarthapati) [inline hub/src/main/java/com/oneday/hub/service/impl/FlightBagServiceImpl.java:71]: since bag service is for outboundservice only wouldnt it be better to call it flightbagservice only, would give a more clear understanding. Also what are these tables we use, is bag manifest genral and for all?
Also shouldnt we rename bag item to flight bag item?

File: hub/src/main/java/com/oneday/hub/service/impl/SortServiceImpl.java [hunk 9]
```diff
@@ -63,4 +82,51 @@ public SortResult resolveOutbound(UUID hubId, ShipmentInfoPort.ParcelInfo parcel
...
+
+        Stand stand = standRepository.findById(bag.getCurrentStandId())
+                .orElseThrow(() -> new StandNotFoundException(bag.getCurrentStandId()));
+
+        eventProducer.emitStandAssigned(parcel.shipmentId(), cityId, hubId,
+                stand.getStandNo(), destHexId.toString(), SortDirection.INBOUND);
+        // The M6 seam: per-parcel sorted-for-delivery feed (M6 binds to a loop) + the M10 leg signal.
+        eventProducer.emitParcelSortedForDelivery(parcel.shipmentId(), cityId, destHexId, validDate,
+                sortedAt, parcel.slaDeadline(), daTerritoryId, routePlanId, loopId, bag.getId(), stand.getStandNo());
+        eventProducer.emitDestSortComplete(parcel.shipmentId(), cityId, hubId, validDate, sortedAt);
```
Comment (Sidarthapati) [inline hub/src/main/java/com/oneday/hub/service/impl/SortServiceImpl.java:126]: Am i emitting all these events together now?
